test/test_utils.py

```diff
@@ -82,12 +82,22 @@ def assert_expected(actual: Any, expected: Any, rtol: float = None, atol: float
     )
 
 
+def tuple_to_dict(t):
+    if not isinstance(t, tuple):
+        raise TypeError(f"Input must be of type tuple but got {type(t)}")
+
+    return {k: v for k, v in enumerate(t)}
+
+
+def is_named_tuple(nt):
+    # hacky way to assert an instance of NamedTuple
```

Comment:
nit: do we have to describe it as "hacky"? As far as I can tell it's the least hacky option available 🙂